Avoid implicit function declarations in tests, for C99 compatibility

Include <stdio.h> for printf and define _GNU_SOURCE for execvpe.

Future compilers will not support implicit function declartions by
default, causing these tests to fail to build.

Submitted upstream: <https://github.com/rizsotto/Bear/pull/521>

diff --git a/test/cases/intercept/preload/errno_reset.c b/test/cases/intercept/preload/errno_reset.c
index c9c82bf6ef727864..82821f3ef0564cdb 100644
--- a/test/cases/intercept/preload/errno_reset.c
+++ b/test/cases/intercept/preload/errno_reset.c
@@ -5,6 +5,7 @@
 #include <stdlib.h>
 #include <errno.h>
 #include <string.h>
+#include <stdio.h>
 
 int main()
 {
diff --git a/test/cases/intercept/preload/posix/execvpe/success.c b/test/cases/intercept/preload/posix/execvpe/success.c
index 0b4b66d0ce16587e..9017be6cfbcc8e3f 100644
--- a/test/cases/intercept/preload/posix/execvpe/success.c
+++ b/test/cases/intercept/preload/posix/execvpe/success.c
@@ -8,6 +8,7 @@
 #include "config.h"
 
 #if defined HAVE_UNISTD_H
+#define _GNU_SOURCE
 #include <unistd.h>
 #endif
 
